### PR TITLE
Fix false positives in PD002

### LINKED ISSUE #4222: Pandas vet incorrectly marks `torch` code as breaking the pandas linting rules
**Issue**:
Pandas vet incorrectly marks `torch` code as breaking the pandas linting rules

**Current result**:
```path/to/file.py:61:35: PD002 `inplace=True` should be avoided; it has inconsistent behavior```

**Expected result**:
No issues raised

**Torch docs**: https://pytorch.org/docs/stable/generated/torch.nn.ReLU.html

**Minimal code/context**:
```!python
    def __init__(
        self,
        <abbreviated>
    ):
        super().__init__()
        <abbreviated>
        self.relu = torch.nn.ReLU(inplace=True)
```
**Command**:
`ruff path/to/file.py`

**Settings**:
```!toml
[tool.ruff]
select = [
    "B", # bugbear
    "C90", # mccabe
    "E", # pycodestyle error
    "F", # pyflakes
    "PD", # pandas vet
    "PL", # pylint
    "RUF", # ruff specific
    "UP", # pyupgrade
    "W", # pycodestyle warning
]
ignore = [
    "B006", # do not use mutable data structures for argument defaults
    "B008", # do not perform function call in argument defaults
    "E731", # lambda-assignment
    "PD901", # bad variable name
    "PLR0911", # too many return statements
    "PLR0912", # too many branches
    "PLR0913", # too many arguments to function call
    "PLR0915", # too many statements
    "PLR2004", # magic numbers
    "PLW2901", # loop variable overwritten by assignment target
]
line-length = 80
target-version = "py37"
```

**Version**:
`ruff 0.0.257`

### PR DESCRIPTION
**Pull request by evanrittenhouse**
Should fix #4222.

### CODE REVIEW

File: crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs [hunk 1]
```diff
@@ -54,10 +55,27 @@ impl Violation for PandasUseOfInplaceArgument {
...
+        is_checkable = true;
+
+        let module = call_path[0];
+        is_pandas = checker.ctx.find_binding(module).map_or(false, |binding| {
+            if let BindingKind::Importation(Importation { full_name, .. }) = binding.kind {
+                full_name == "pandas"
+            } else {
+                false
+            }
+        });
```
Comment (MichaReiser) [inline crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs:76]: Nit
```suggestion
        is_pandas = checker.ctx.find_binding(module).map_or(false, |binding| {
            matches!(
                binding.kind,
                BindingKind::Importation(Importation {
                    full_name: "pandas",
                    ..
                })
            )
        });
```

File: crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs [hunk 2]
```diff
@@ -98,6 +116,15 @@ pub fn inplace_argument(
...
                     }
                 }
+
+                // Without a static type system, only module-level functions could potentially be
+                // non-pandas calls. If they're not, `inplace` should be considered safe.
+                if is_checkable {
+                    if !is_pandas {
+                        return None;
+                    }
+                }
```
Comment (MichaReiser) [inline crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs:126]: Nit
```suggestion
                if is_checkable && !is_pandas {
                    return None;
                }
```

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs (hunk 1 above)
Comment (evanrittenhouse) [reply crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs:76]: Done - thanks!

**Comment by MichaReiser** [comment]
Thanks for another awesome contribution :)

**Pull request merged**